cmd/controller/main.go

```diff
@@ -27,11 +29,22 @@ func main() {
 	flag.BoolVar(&ctrlOpts.EnablePprof, "dangerous-enable-pprof", false, "If set to true, enable pprof on "+PprofListenAddr)
 	flag.DurationVar(&ctrlOpts.APIRequestTimeout, "api-request-timeout", time.Duration(0), "HTTP timeout for Kubernetes API requests")
 	flag.BoolVar(&ctrlOpts.APIPriorityAndFairness, "enable-api-priority-and-fairness", true, "Enable/disable APIPriorityAndFairness feature gate for apiserver. Recommended to disable for <= k8s 1.19.")
-	flag.BoolVar(&sidecarexec, "sidecarexec", false, "Run sidecarexec")
+	flag.BoolVar(&isSidecarexec, "sidecarexec", false, "Run sidecarexec")
+	flag.BoolVar(&isSidecarexecDebug, "sidecarexecdebug", false, "Run sidecarexecdebug")
+	flag.BoolVar(&isSidecarexecWrap, "sidecarexecwrap", false, "Run sidecarexecwrap")
 	flag.Parse()
 
-	if sidecarexec {
-		sidecarexecMain()
+	if isSidecarexec || isSidecarexecDebug {
+		sidecarexecMain(isSidecarexecDebug, flag.Args())
+		return
+	}
+	if isSidecarexecWrap {
```

Comment:
What's the `wrap` part for? When would I choose `sidecarexecwrap` over `sidecarexec` seen as they are exclusive

Reply:
sidecarexecwrap flag is used internally by https://github.com/vmware-tanzu/carvel-kapp-controller/pull/785/files#diff-8289313d09c3e91a49ad1d1a7230003f8b93e1b75dbdbe2cc1454b31e51d0956R45. 

effectively sidecarexec server creates a new sandboxed process (linux namespaces applied), and wrap command is what gets executed to continue sandboxing process (it does mounting of directories and pivot_root) before executing the actual command we care about.

may be a different name would make this a bit better? sidecarexecCmdInit?

Reply:
yeah I got a lil confused with the naming at first tbh

As of right now, in my head:
`sidecarexec` == A container that is a RPC server that allows a whitelist of commands run in an isolated way.

Therefore, `sidecarexecCmdInit` to me, sounds like the Init command for the above container.

my 2 cents:
- keep it sidecarexecwrap
- sidecarexecLiteChroot
- sidecarexecJail
- sidecarexecVirtualEnv
- sidecarexecLiteContainer

lol naming is hard... not convinced myself with the above. posting anyway.

Reply:
another option sidecarexecPreExec